frontend/src/PortfolioInfo.jsx

```diff
@@ -0,0 +1,195 @@
+      });
+      if (response.ok) {
+        let data = await response.json();
+        newArray.push({ id: data.id, name: data.name, ticker: data.ticker });
+      }
+    }
+    setCompaniesData(newArray);
+    const prices = [];
+    for (const company of newArray) {
+      const stockResponse = await fetch(
+        `${BASE_URL}/getters/stats/${company.ticker}`
+      );
+      const stockData = await stockResponse.json();
+      prices.push({
+        price: stockData.regularMarketPrice,
+        dayStart: stockData.regularMarketPreviousClose,
+      });
+    }
+    setCompaniesStockData(prices);
+  };
+  useEffect(() => {
+    getPortfolioData();
+    getCompaniesData();
+  }, [companyIds]);
+
+  if (portfolioData == null) {
+    return (
+      <img
+        className="w-40 h-40 mt-10 ml-auto mr-auto"
+        src="https://i.gifer.com/ZKZg.gif"
+      />
+    );
+  }
+
+  const handleClick = (id) => {
+    navigate(`/CompanyInfo/${id}`);
+  };
+
+  return (
+    <>
+      <SidebarProvider>
+        <AppSidebar />
+        <div className="relative h-full flex flex-col justify-center">
+          <SidebarTrigger className="fixed top-1/2" />
+        </div>
+        <header className="bg-indigo-50 fixed top-0 w-full pt-4 flex flex-col items-center h-16">
+          <h2 className="fixed top-0 left-4 text-4xl pt-3 font-medium">
+            Alpha-Edge
+          </h2>
+          <h4 className="fixed top-0 right-4 pt-5">
+            Good day, <span className="font-bold"> {fullName} </span>
+          </h4>
+        </header>
+        <main className="w-full">
+          <div className="flex flex-col items-center">
+            <SearchBar />
+            <h3 className="self-center text-center text-6xl mt-30 mb-10 text-indigo-50 font-semibold drop-shadow-[0px_0px_39px_rgba(247,247,247,.3)] z-10">
+              {portfolioData.name}
+            </h3>
+            <div className="flex flex-row justify-start w-full">
+              <div className="bg-indigo-50 h-75 w-7/20 ml-40 rounded-md overflow-auto">
+                <div className="flex flex-row justify-center">
+                  <h3 className="font-bold text-2xl text-center mt-3">
+                    Portfolio Companies
+                  </h3>
+                  <Pencil
+                    className="mt-5 ml-4 border-2 h-5 w-5 rounded-sm hover:scale-110 transition-transform duration-300 ease-in-out hover:cursor-pointer hover:brightness-90"
+                    onClick={() => {
+                      setIsEditingMode((prev) => !prev);
+                    }}
+                  />
+                </div>
+                <div className="flex flex-col h-full items-center">
+                  {companiesStockData.length === 0 && (
+                    <img
+                      className="h-20 w-20 mt-10"
+                      src="https://i.gifer.com/ZKZg.gif"
+                    />
+                  )}
+                  {companiesStockData.length !== 0 &&
```

Comment:
Whoa, this is getting pretty nested -- you might want to refactor the below into a component for an individual company's data for readability.